Realm/Realm/Realm.cs

```diff
@@ -557,13 +557,13 @@ internal RealmMetadata MergeSchema(RealmSchema schema)
         /// </remarks>
         /// <returns>The passed object, so that you can write <c>var person = realm.Add(new Person { Id = 1 });</c>.</returns>
         public T Add<T>(T obj, bool update = false)
-            where T : RealmObject
+            where T : IRealmObject
         {
             ThrowIfDisposed();
             Argument.NotNull(obj, nameof(obj));
 
             // This is not obsoleted because the compiler will always pick it for specific types, generating a bunch of warnings
-            AddInternal(obj, typeof(T), update);
+            AddInternal(obj, obj.GetType(), update);
```

Comment:
Why is this change necessary?

Reply:
It was an error

Reply:
The problem here is doing something like: `realm.Add((RealmObject)Activator.CreateInstance(type));`
With the previous code, this will call the alternative `Add` method that already uses `obj.GetType()`.  
So, either we keep `obj.GetType()` here too, or we need to keep both `Add` (with `IRealmObject` and `RealmObject`)

(disregard the fact that this is outdated, I was trying something)

Reply:
@nirinchev (for visibility, given it says it's outdated due to some changes I've made)